src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -7232,6 +7503,42 @@ final void trySetSensitivityClassification(SensitivityClassification sensitivity
 }
 
 
+/**
+ * The tds default implementation of a timeout command
+ */
+class TdsTimeoutCommand extends TimeoutCommand<TDSCommand> {
+    protected TdsTimeoutCommand(int timeout, TDSCommand command, SQLServerConnection sqlServerConnection) {
+        super(timeout, command, sqlServerConnection);
+    }
+
+    protected void interrupt() throws Exception {
+        TDSCommand command = getCommand();
+        SQLServerConnection sqlServerConnection = getSqlServerConnection();
+        try {
+            // If TCP Connection to server is silently dropped, exceeding the query timeout
+            // on the same connection does
+            // not throw SQLTimeoutException
+            // The application stops responding instead until SocketTimeoutException is
+            // thrown. In this case, we must
+            // manually terminate the connection.
+            if (null == command && null != sqlServerConnection) {
+                sqlServerConnection.terminate(SQLServerException.DRIVER_ERROR_IO_FAILED,
+                        SQLServerException.getErrString("R_connectionIsClosed"));
+            } else {
+                // If the timer wasn't canceled before it ran out of
+                // time then interrupt the registered command.
+                command.interrupt(SQLServerException.getErrString("R_queryTimedOut"));
+            }
+        } catch (SQLServerException e) {
+            // Request failed to time out and SQLServerConnection does not exist
+            if (null != command)
+                command.log(Level.FINE, "Command could not be timed out. Reason: " + e.getMessage());
+            throw new Exception("Command could not be timed out. Reason: " + e.getMessage());
```

Comment:
Since this exception can be surfaced to the application and thus the user, this string needs to go into resources so that it can be localized.

Reply:
Will be updated in follow-up PR.